### PR TITLE
Fixes: Translations Are Not Correctly Tested in Our PR Workflow 

### LINKED ISSUE #2260: Translations Are Not Correctly Tested in Our PR Workflow
**Describe the bug**

1. Multilingual support is an important feature of our application.
2. We need to ensure that we have translations created for every language entry
3. We created a script to test for this in Talawa mobile to test for this. It was added to this repo, but doesn't appear to work.
    1. https://github.com/PalisadoesFoundation/talawa/pull/2284
1. We need to apply the script to this repository and make adjustments where necessary. Ideally the script in both repos must be identical and apply the logic to different directory trees. Make updates accordingly.
1. Without correct translations, the end user UI/UX is unacceptable.

**To Reproduce**

- See the **Actual behavior** section below

**Expected behavior**

1. We need to create a GitHub action that during the PR process checks all the translation files to make sure that there are translations in all the non-default languages for each English default language entry.
1. As part of the PR for this issue, all the translations must be correctly present

**Actual behavior**

1. This PR failed to notice the incomplete translations
    1. https://github.com/PalisadoesFoundation/talawa-admin/pull/2254
1. Only one translation file was edited
    image
1. There are no errors in the PR workflow
    image
    image

**Screenshots**

- See above

**Additional details**

- N/A

**Potential internship candidates**

Please read this if you are planning to apply for a Palisadoes Foundation internship
- https://github.com/PalisadoesFoundation/talawa/issues/359

### PR DESCRIPTION
**Pull request by Nandika-A**
**What kind of change does this PR introduce?**

**Issue Number:**

Fixes #2260 

**Did you add tests for your changes?**
No

**Snapshots/Videos:**

image

**If relevant, did you update the documentation?**
No

**Summary**
1. Modified compare_translations.py to handle nested JSON objects correctly. The previous implementation only compared top-level keys, causing issues with missing or extra keys in nested structures.
2. Added the function "flatten_json" to recursively compare keys for nested json.
3. Error messages now show the full key path using dot notation .

**Does this PR introduce a breaking change?**
No

**Other information**

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-admin/blob/master/CONTRIBUTING.md)?**

Yes

## Summary by CodeRabbit

- **New Features**
	- Introduced a function to flatten nested JSON objects, improving data usability.
	- Updated translation loading to return flattened data for easier handling.
	- Expanded localization capabilities with numerous new key-value pairs in English, Hindi, Chinese, French, and Spanish, enhancing user experience across various functionalities.

- **Bug Fixes**
	- Adjusted the default directory path for translation files to ensure correct file access.

### CODE REVIEW

**Review by palisadoes — APPROVED** [review]

**Review by palisadoes — CHANGES_REQUESTED** [review]
Please add the missing translations in this PR or else all new PRs will fail

**Comment by Nandika-A** [comment]
> Please add the missing translations in this PR or else all new PRs will fail

Okay sure

**Comment by palisadoes** [comment]
1. Please fix the failing tests. The translation check either isn't working correctly or you have missing translations.
2. We cannot merge this PR until the checks are fixed or else all other PRs will fail before the code base is tested.

**Comment by Nandika-A** [comment]
> 1. Please fix the failing tests. The translation check either isn't working correctly or you have missing translations.
> 2. We cannot merge this PR until the checks are fixed or else all other PRs will fail before the code base is tested.

Yes, two of the files were missing some translations. I have added them and no other translations are remaining now. Could you please review them?

**Comment by palisadoes** [comment]
Please fix the failing tests.

File: src/screens/OrganizationPeople/AddMember.tsx
[the remark this answers is not in the document]
Comment (Nandika-A) [reply src/screens/OrganizationPeople/AddMember.tsx:1]: These changes got included when I merged this branch with develop.

[the remark this answers is not in the document]
Comment (Nandika-A) [reply src/screens/OrganizationPeople/AddMember.tsx:1]: Okay sure!

**Comment by Nandika-A** [comment]
Hi @palisadoes ! I have made all the changes, could you please review them?

**Review by palisadoes — APPROVED** [review]

**Pull request merged**